### PR TITLE
Merge 2.3 into master

### PR DESCRIPTION
**Pull request by mrclay**

### CODE REVIEW

**Comment by mrclay** [comment]
Some fatal error in simpletests.

File: engine/classes/ElggEntity.php [hunk 1]
```diff
@@ -1601,22 +1601,28 @@ public function disable($reason = "", $recursive = true) {
...
-			]);
-			$subentities->setIncrementOffset(false);
+				'limit' => false,
+			];
 			
-			foreach ($subentities as $subentity) {
-				/* @var $subentity \ElggEntity */
-				if (!$subentity->isEnabled()) {
-					continue;
+			foreach (['owner_guid', 'container_guid', 'site_guid'] as $db_column) {
```
Comment (jdalsem) [inline engine/classes/ElggEntity.php]: remove site_guid

File: engine/classes/ElggEntity.php [hunk 2]
```diff
@@ -1773,21 +1779,26 @@ public function delete($recursive = true) {
...
-			foreach ($batch as $e) {
-				$e->delete(true);
+			$base_options = [
+				'wheres' => [
+					"e.guid != $guid",
+				],
+				'limit' => false,
+			];
+			
+			foreach (['owner_guid', 'container_guid', 'site_guid'] as $db_column) {
```
Comment (jdalsem) [inline engine/classes/ElggEntity.php]: remove site_guid

**Review by jdalsem — CHANGES_REQUESTED** [review]

**Comment by jdalsem** [comment]
maybe my comments solve the simplecache fails

**Comment by jdalsem** [comment]
New version here: #10837

**Pull request closed without merging**